packages/std/src/math/decimal.rs

```diff
@@ -184,16 +184,26 @@ impl Decimal {
         Self::DECIMAL_PLACES as u32
     }
 
+    /// Rounds value down after decimal places.
     pub fn floor(&self) -> Self {
         Self((self.0 / Self::DECIMAL_FRACTIONAL) * Self::DECIMAL_FRACTIONAL)
     }
 
+    /// Rounds value up after decimal places. Panics on overflow.
     pub fn ceil(&self) -> Self {
+        match self.checked_ceil() {
+            Ok(value) => value,
+            Err(_) => panic!("attempt to ceil with overflow"),
+        }
+    }
+
+    /// Rounds value up after decimal places. Returns OverflowError on overflow.
+    pub fn checked_ceil(&self) -> Result<Self, OverflowError> {
         let floor = self.floor();
         if &floor == self {
-            floor
+            Ok(floor)
         } else {
-            floor + Decimal::one()
+            floor.checked_add(Decimal::one())
```

Comment:
Hmm, I think the overflow error containd values as payload that are more misleading than helping as they refer to implementation details.

Can we just use `Option` instead or create a new error witout the payoad?

Reply:
I like the option idea, but right now it would be inconsistent with the rest of the types. IMO it should be changed in a bunch for all types.
I'll create a new error then.

Reply:
Sounds good, thanks!